osquery/tables/system/linux/processes.cpp

```diff
@@ -56,6 +58,32 @@ inline std::string readProcCMDLine(const std::string& pid) {
   return content;
 }
 
+inline std::string readProcCgroup(const std::string& pid) {
+  auto attr = getProcAttr("cgroup", pid);
+
+  std::string content;
+  readFile(attr, content);
```

Comment:
Naive question... Is there a performance impact from there? I think we use the process table for a variety of internal monitoring, right? And this would add a proc read per process per query.

Reply:
Yes, this will take more time in order to do a proc read. The processes table already has several other proc reads. I'd want some way to lazily execute the code but without that we're stuck between "an extra proc read for every process" or "another table with a join". Looking at [a blog post about syscall time](https://gms.tf/on-the-costs-of-syscalls.html) it seems like we should be adding on the order of 2µs per process, though I haven't tested this.

Reply:
Thanks to the latest changes applied, I think the potential impact on performances has been reduced. Here's a (rough) quick test with `.timer on`:

processes
 * 21463 processes on the host
 * 2046 of which are from a container
 * runtime used: libpod
 * 10 runs on each branch

cgroup paths had the following values:
 * On Fedora, most processes will return cgroup_path as '/'
 * 66 had `/system.slice/%`
 * 93 had `/user.slice/%`
 * 2046 had `/machine.slice/libpod-%`

offsets
 * real +9.48%
 * user +8.82%
 * sys +12.40%

Perf should be able to provide better readings (and a diff) if required

Reply:
Latest patch should have no measurable perf impact if `cgroup_path` is not used

Reply:
Thanks for updating this